src/contents/util/courseLogic.ts

```diff
@@ -237,31 +236,23 @@ const splitByNumbers = (input: string): string[] => {
 
 export const createSyllabusButton = async () => {
   if (!location.href.startsWith("https://scombz.shibaura-it.ac.jp/lms/course?")) return;
-  const currentData = (await chrome.storage.local.get(defaultSaves)) as Saves;
-  const fac = currentData.settings.faculty;
+
   const insertArea =
     document.querySelector(".contents-question-template-area") || document.querySelector(".course-title-txt");
-  if (!fac) {
-    insertArea.insertAdjacentHTML(
-      "afterend",
-      `<span style="color:red;padding: 12px 30px 10px 34px">
-        ${chrome.i18n.getMessage("openExtensionOptionDescription1")}
-        <a href="javascript:void(0);" id="link_to_extention_syll">${chrome.i18n.getMessage("extensionOption")}</a>
-        ${chrome.i18n.getMessage("openExtensionOptionDescription2")}
-        </span>
-    `,
-    );
-    document.getElementById("link_to_extention_syll")?.addEventListener("click", () => {
-      chrome.runtime.sendMessage({ action: "openOption" } as RuntimeMessage);
-    });
-    return;
-  }
+
+  const year: string = location.search.split("idnumber=")[1].slice(0, 4);
+  const facNumber: number = parseInt(location.search.split("idnumber=")[1].slice(4, 6), 10);
+  const facArray = ["ko1", "arc", "sys", "dsn", "din"];
+  const fac = Number.isNaN(facNumber) ? undefined : facArray[facNumber - 1];
+  const idxnameArray = facArray.map((fac) => `&idxname=${fac}${year}`);
+  const idxname = fac ? idxnameArray[facArray.indexOf(fac)] : idxnameArray.join("");
```

Comment:
`&idxname=${year}%2F${fac}`じゃないとうまく検索されないですね。
また、`facArray.indexOf(fac)`は`facNumber - 1`であることが自明ですね。
それと今気づきましたが、`facArray[NaN]`も`undefined`を返すので`Number.isNaN(facNumber)`で判定する意味ないですね。

```suggestion
  const fac = facArray[facNumber - 1];
  const idxnameArray = facArray.map((fac) => `&idxname=${year}%2F${fac}`);
  const idxname = fac ? idxnameArray[facNumber - 1] : idxnameArray.join("");
```

Reply:
> `&idxname=${year}%2F${fac}`じゃないとうまく検索されないですね。 
お恥ずかしいことに脱字してました…

> \また、mapの引数として、大きいスコープで使われている変数と同じ変数名をとらないでください。
了解です。